docs/setup/keadm.md

```diff
@@ -96,70 +96,273 @@ KubeEdge edgecore is running, For logs visit:  /var/log/kubeedge/edgecore.log
+   Run the following command on the host on which each apiserver runs:
+   
+    ** Note: ** You need to set the cloudcoreips variable first
+
+    ```bash
+    iptables -t nat -A OUTPUT -p tcp --dport 10350 -j DNAT --to $CLOUDCOREIPS:10003
+    ```
+    > Port 10003 and 10350 are the default ports for the CloudStream and edgecore,
+      use your own ports if you have changed them.
+  
+    If you are not sure if you have setting of iptables, and you want to clean all of them.
+    (If you set up iptables wrongly, it will block you out of your `kubectl logs` feature)
+    The following command can be used to clean up iptables:
+    ``` shell
+    iptables -F && iptables -t nat -F && iptables -t mangle -F && iptables -X
+    ```
+
+5. Modify **both** `/etc/kubeedge/config/cloudcore.yaml` and `/etc/kubeedge/config/edgecore.yaml` on cloudcore and edgecore. Set up **cloudStream** and **edgeStream** to `enable: true`. Change the server IP to the cloudcore IP (the same as $CLOUDCOREIPS).
+
+    Open the YAML file in cloudcore:
+    ```shell
+    sudo nano /etc/kubeedge/config/cloudcore.yaml
+    ``` 
+
+    Modify the file in the following part (`enable: true`):
+    ```yaml
     cloudStream:
       enable: true
-  ...
-
-  ...
-  modules:
+      streamPort: 10003
+      tlsStreamCAFile: /etc/kubeedge/ca/streamCA.crt
+      tlsStreamCertFile: /etc/kubeedge/certs/stream.crt
+      tlsStreamPrivateKeyFile: /etc/kubeedge/certs/stream.key
+      tlsTunnelCAFile: /etc/kubeedge/ca/rootCA.crt
+      tlsTunnelCertFile: /etc/kubeedge/certs/server.crt
+      tlsTunnelPrivateKeyFile: /etc/kubeedge/certs/server.key
+      tunnelPort: 10004
+    ```
+
+    Open the YAML file in edgecore:
+    ``` shell
+    sudo nano /etc/kubeedge/config/edgecore.yaml
+    ```
+    Modify the file in the following part (`enable: true`), (`server: 192.168.0.193:10004`):
+    ``` yaml  
     edgeStream:
       enable: true
-  ...
-  ```
+      handshakeTimeout: 30
+      readDeadline: 15
+      server: 192.168.0.139:10004
+      tlsTunnelCAFile: /etc/kubeedge/ca/rootCA.crt
+      tlsTunnelCertFile: /etc/kubeedge/certs/server.crt
+      tlsTunnelPrivateKeyFile: /etc/kubeedge/certs/server.key
+      writeDeadline: 15
+    ```  
 
 6. Restart all the cloudcore and edgecore.
 
+    ``` shell
+    sudo su
+    ```
+    cloudCore:
+    ``` shell
+    pkill cloudcore
+    nohup cloudcore > cloudcore.log 2>&1 &
+    ```
+    edgeCore:
+    ``` shell
+    systemctl restart edgecore.service
+    docker ps
+    ```
+    | CONTAINER ID  | NAMES                                           |
+    |---------------|-------------------------------------------------|
+    | d40ad4007d6f9 | k8s_POD_weave-net-fr9vx_kube-system_...         |
+    | f59bdacbfc48  | k8s_kube-proxy_kube-proxy-s8476_kube-system_... |
+    | f15a4b9aae7a  | k8s_POD_kube-proxy-s8476_kube-system_.....      |
+
+
+    Kill those `Network` related containers to make restart 
```

Comment:
Do we need to restart all thses containers?

Reply:
@fisherxu 
In the general case, it is essentially important, unless the user has made some modification of their **edgecore.service** file or **kubeproxy daemonset yaml** file.

Reply:
So I'm not sure why need to restart these proxy and cni releated containers?

Reply:
@fisherxu 
If I don't kill those containers first, the restart of edgecore will fail.

Reply:
Thanks for the clarification, but I don't think we need to restart these containers.

If users want to upgrade or restart the edgecore, and the users apps must not be restart.

> If I don't kill those containers first, the restart of edgecore will fail.

What's the relationship between those containers and edgecore?